samples/nfc/tag_reader/src/main.c

```diff
@@ -0,0 +1,192 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <misc/printk.h>
+
+#include <drivers/st25r3911b/nfca.h>
+
+#define NFC_T2T_READ_CMD 0x30
+#define NFC_BLOCK_TO_READ_NUM 0
+
+#define NFC_T2T_READ_CMD_LEN 0x02
+#define NFC_T2T_BLOCK_LEN 0x10
+
+#define NFC_TX_DATA_LEN NFC_T2T_READ_CMD_LEN
+#define NFC_RX_DATA_LEN NFC_T2T_BLOCK_LEN
+
+#define TRANSMIT_DELAY 3000
+#define ALL_REQ_DELAY 2000
+#define T2T_READ_CMD_FDT 2068
+
+u8_t tx_data[NFC_TX_DATA_LEN] = {NFC_T2T_READ_CMD, NFC_BLOCK_TO_READ_NUM};
+u8_t rx_data[NFC_RX_DATA_LEN];
+
+static struct k_delayed_work transmit_work;
+
+struct nfca_buf tx_buf = {
+		.data = tx_data,
+		.len = sizeof(tx_data)
+};
+
+struct nfca_buf rx_buf = {
+		.data = rx_data,
+		.len = sizeof(rx_data)
+};
+
+static void transfer_handler(struct k_work *work)
+{
+	int err = st25r3911b_nfca_tag_detect(DETECT_CMD_SENS_REQ);
+
+	if (err) {
+		printk("Tag detect error: %d\n", err);
+		return;
+	}
+}
+
+static void nfc_field_on(void)
+{
+	printk("NFC field on\n");
+
+	int err = st25r3911b_nfca_tag_detect(DETECT_CMD_SENS_REQ);
+
+	if (err) {
+		printk("Tag select error: %d\n", err);
+		return;
+	}
+}
+
+static void nfc_timeout(bool tag_sleep)
+{
+	int err;
+
+	if (tag_sleep) {
+		printk("Tag sleep or no detected, sending ALL Request,\n");
+	}
+
+	k_sleep(ALL_REQ_DELAY);
```

Comment:
I am not fond of this sleep as it is performed in callback... it will work as callback is called from a thread but anyway

Reply:
Ok. I can use a delayed workqueue to handle this.

Reply:
I think it's better to leave it. It's sample and adding work will just complicate things.
`/* Sleep will block processing loop. Accepted as it is short. */`